### PR TITLE
Added scale and resizeProportionally methods.

### PR DESCRIPTION
**Pull request by VVelda**
I know, there is snippet `$image->resize($image->getWidth() * .5)` to mimic scale function, but it would be convenient to have some method with semantic.
Also added resizeProportionally method, which is somehow similar to bestFit, but final image have always specified dimensions with some background color.

### CODE REVIEW

File: src/claviska/SimpleImage.php [hunk 1]
```diff
@@ -801,6 +801,60 @@ public function resize($width = null, $height = null) {
...
+    
+    $aspectRatio = $width / $height;
+    if($aspectRatio > $this->getAspectRatio())
+    $this->resize(null, $height);
+    else
+    $this->resize($width, null);
+    
+    // fill background, then place centered resized image
+    $newImage = imagecreatetruecolor($width, $height);
+    fwrite(STDERR, "test: " . $color['red'] . " " . $color['green'] . " " . $color['blue'] . " " . $color['alpha']);
```
Comment (claviska) [inline src/claviska/SimpleImage.php]: This needs to be removed.

File: src/claviska/SimpleImage.php [hunk 2]
```diff
@@ -801,6 +801,60 @@ public function resize($width = null, $height = null) {
...
+  //
+  public function resizeProportionally($width, $height = null, $backgroundColor = 'transparent') {
+    // $backgroundColor = $this->allocateColor($backgroundColor);
+    $color = $this->normalizeColor($backgroundColor);
+    
+    if(!$height)
+    $height = $width;
+    
+    $aspectRatio = $width / $height;
+    if($aspectRatio > $this->getAspectRatio())
```
Comment (claviska) [inline src/claviska/SimpleImage.php]: Conditional formatting is inconsitent. Please use brackets :)

**Comment by claviska** [comment]
Thanks for the thorough PR! I marked up a couple things I noticed in the code.

I really think creating a new SimpleImage instance and using `overlay()` will be better than using raw image methods. For example:

```php
$newImage
  ->fromNew($width, $height, $color)
  ->overlay($originalImage);
```

The `overlay()` method works with transparent GIFs and PNGs, so it will prevent some edge cases where transparency gets messed up. (It uses [`imageCopyMergeAlpha`](https://github.com/claviska/SimpleImage/blob/master/src/claviska/SimpleImage.php#L447) instead of [`imagecopy`](http://php.net/manual/en/function.imagecopy.php) to do this.)

File: src/claviska/SimpleImage.php [hunk 3]
```diff
@@ -801,6 +801,60 @@ public function resize($width = null, $height = null) {
...
+  //  $factor* (float) - Factor of the scale.
+  //
+  // Returns a SimpleImage object.
+  //
+  public function scale($factor) {
+    return $this->resize($this->getWidth() * $factor);
+  }
+
+  //
+  // Resize an image to tightly fit a rectangular of the specified dimensions, without deformation.
```
Comment (claviska) [inline src/claviska/SimpleImage.php]: "rectangular" and "centred" typo

**Comment by VVelda** [comment]
Hello! I've did some requested changes.

I do not change requestProportionally implementations as this is working – I have tested transparent img on transparent bg, nontransparent on transparent and transparent on nontransparent. You can verify with example/index_mod.php.

**Comment by claviska** [comment]
The point is to eliminate duplicate code, especially code that’s prone to edge cases. Transparent images in GD are a pain. The overlay method is a battle-tested way to merge two images.

In its current state, when new edge cases come up, we’ll need to fix them in multiple methods. This means extra work and leaves room for error. By using existing methods, we’ll only have to fix those types of things once. 😄

**Comment by claviska** [comment]
Closing this due to lack of activity.

**Pull request closed without merging**